web/src/site-admin/SiteAdminActivation.tsx

```
@@ -0,0 +1,259 @@
+            }
+            repositories(enabled: true) {
+                totalCount
+            }
+            viewerSettings {
+                final
+            }
+        }
+    `).pipe(
+        map(dataOrThrowErrors),
+        map(data => {
+            const settings = JSON.parse(data.viewerSettings.final)
+            const authProviders = window.context.authProviders
+            return {
+                connectedCodeHost: data.externalServices.totalCount > 0,
+                enabledRepository: data.repositories.totalCount !== null && data.repositories.totalCount > 0,
+                enabledExtension:
+                    settings.extensions && Object.values(settings.extensions).filter(enabled => enabled).length > 0,
+                enabledSignOn: !!(authProviders && authProviders.filter(p => !p.isBuiltin).length > 0),
+            }
+        })
+    )
+
+type SiteAdminActivationPopoverButtonProps = CommandListProps & { history: H.History }
+interface SiteAdminActivationPopoverButtonState {
+    checklistInfo?: SiteAdminChecklistInfo
+}
+
+/**
+ * SiteAdminActivationPopoverButton is the nav bar button that opens the site admin
+ * activation checklist as a dropdown.
+ */
+export class SiteAdminActivationPopoverButton extends React.PureComponent<
+    SiteAdminActivationPopoverButtonProps,
+    SiteAdminActivationPopoverButtonState
+> {
+    private subscriptions = new Subscription()
+
+    constructor(props: SiteAdminActivationPopoverButtonProps) {
+        super(props)
+        this.state = {
+            checklistInfo: {
+                connectedCodeHost: false,
+                enabledRepository: false,
+                enabledExtension: false,
+                enabledSignOn: false,
+            },
+        }
+    }
+
+    public componentDidMount(): void {
+        this.subscriptions.add(
+            fetchSiteAdminChecklist().subscribe(checklistInfo => {
+                this.setState({ checklistInfo })
+            })
```

This swallows the error if the request fails. The user should be informed that something went wrong.